final/introduction.tex

```diff
@@ -1,6 +1,33 @@
 \chapter{Introduction}
 \label{cha:introduction}
 
+Besides general purpose languages such as Java and C, nowadays there are also a
+lot of domain specific languages available. Domain specific languages can
+provide a lot of expressive power in a particular problem domain, thereby
```

Comment:
And make domain-specific optimizations, plus reduce the surface for problems due to allowing only certain constructions etc. Perhaps worth mentioning.